test/firewall_stage1_finish_test.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+require "yast"
+require_relative "../src/clients/firewall_stage1_finish"
+
+describe Yast::FirewallStage1FinishClient do
+  describe "main" do
+    context "when the client is called with 'Info'" do
+      it "returns a hash with 'steps', 'title' and 'when' keys" do
+
+        res = Yast::WFM.CallFunction("firewall_stage1_finish", ["Info"])
+
+        expect(res).to be_a(Hash)
+        expect(res).to include("steps", "title", "when")
+      end
+    end
+
+    context "when the client is called with 'Write'" do
+      before do
+        allow(Yast::WFM).to receive(:Args).and_return("Write")
+      end
+
+      context "in autoninst Mode" do
+        let(:installed) { false }
+
+        before do
+          allow(Yast::Mode).to receive(:autoinst).and_return(true)
+          allow(Yast::SuSEFirewall).to receive(:WriteConfiguration)
+          allow(Yast::Service).to receive(:Enable)
+          allow(Yast::SuSEFirewall4Network).to receive(:IsInstalled).and_return(installed)
+          allow(subject).to receive(:adjust_ay_configuration)
+        end
+
+        context "with SuSEfirewall2 installed" do
+          let(:installed) { true }
+
+          it "allows the remote installations in use in the SuSEFirewall configuration" do
+            expect(subject).to receive(:adjust_ay_configuration)
+
+            subject.main
+          end
+        end
+
+        context "without SuSEFirewall installed" do
+          it "does nothing special for AutoYaST" do
+            expect(subject).to_not receive(:adjust_ay_configuration)
+
+            subject.main
+          end
+        end
+      end
+
+      it "enables the sshd service in case of ssh remote installation" do
+        expect(Yast::SuSEFirewall4Network).to receive(:EnabledSshd).and_return(true)
+        expect(Yast::Service).to receive(:Enable).with("sshd")
+
+        subject.main
+      end
+
+      it "writes the SuSEFirewall configuration" do
+        expect(Yast::SuSEFirewall).to receive(:WriteConfiguration)
+
+        subject.main
+      end
+    end
+  end
+
+  describe "#adjust_ay_configuration" do
+    before do
+      allow(Yast::Progress).to receive(:set)
+      allow(Yast::SuSEFirewall).to receive(:Read)
+      allow(subject).to receive(:open_ssh_port)
+      allow(subject).to receive(:open_vnc_port)
+      allow(Yast::Linuxrc).to receive(:useiscsi).and_return(false)
+      allow(Yast::SuSEFirewall4Network).to receive(:Enabled1stStage).and_return(true)
+    end
+
+    context "when the user has not modified the proposal" do
+      it "obtains the linuxrc options and defauls from the control file" do
+        allow(Yast::SuSEFirewallProposal).to receive(:GetChangedByUser).and_return(false)
```

Comment:
*Warning: just a matter of preference. Feel free to ignore. :)*

Given that you already have a different context, I would move the `allow` into a before block. Looks more readable to me. Same below.